packages/nuxt/src/core/nuxt.ts

```diff
@@ -625,6 +625,14 @@ async function initNuxt (nuxt: Nuxt) {
       filePath: resolve(nuxt.options.appDir, 'components/nuxt-island'),
     })
 
+    // sync conditions with /packages/nuxt/src/core/templates.ts#L539
+    if (nuxt.options.dev || nuxt.options.experimental.componentIslands !== 'auto' || nuxt.apps.default?.pages?.some(p => p.mode === 'server') || nuxt.apps.default?.components?.some(c => c.mode === 'server' && !nuxt.apps.default?.components.some(other => other.pascalName === c.pascalName && other.mode === 'client'))) {
```

Comment:
actually, we may need to move the addition of this event handler later as `nuxt.apps.default` isn't yet initialised.